gradle-plugin/src/main/java/com/squareup/anvil/plugin/CheckMixedSourceSet.kt

```diff
@@ -1,65 +0,0 @@
-package com.squareup.anvil.plugin
-
-import org.gradle.api.GradleException
-import org.gradle.api.Project
-import org.gradle.api.plugins.JavaPluginConvention
-import org.jetbrains.kotlin.gradle.internal.Kapt3GradleSubplugin
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-import java.io.File
-
-/**
- * In a mixed Kotlin / Java source set the Kotlin compiler might crash with an error like this:
- *
- * java.lang.AssertionError: Duplicated JavaClassDescriptor a/b/c/Hello reported to IC.
- *
- * That's a known bug: https://youtrack.jetbrains.com/issue/KT-38576
- *
- * The workaround for now is to set the kotlin.incremental.usePreciseJavaTracking flag to false for
- * these module using this task.
- */
-open class CheckMixedSourceSet(
```

Comment:
The change itself is good. But I really dislike that you deleted this class and moved everything into the plugin class. You bloat the plugin class even more. I suggest to keep this class simply move the logic here. This is just a question of how to organize code.

You also removed the comment from this class which is very valuable to understand why we do this at all.